mk/spksrc.service.installer

```diff
@@ -145,6 +150,39 @@ set_syno_permissions ()
     fi
 }
 
+# Sets recurivse permissions using chown
+set_unix_permissions ()
+{
+    DIRNAME=$1
+    echo "Granting '${EFF_USER}' unix permissions on ${DIRNAME}" >> ${INST_LOG}
+    if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
+        chown -R ${EFF_USER}:root "${DIRNAME}" >> $INST_LOG 2>&1
+    else
+        chown -R ${EFF_USER}:${USER} "${DIRNAME}" >> $INST_LOG 2>&1
+    fi
+}
+
+# If package was moved to new group, we need to add the new package user
+# also to the old group. Only if the legacy user was in the old group.
+# Usage: syno_user_add_to_legacy_group "${NEW_USER}" "${LEGACY_USER}" "${LEGACY_GROUP}"
+syno_user_add_to_legacy_group () {
+    NEW_USER=$1
+    LEGACY_USER=$2
+    LEGACY_GROUP=$3
+
+    # Check if user in old group
+    MEMBERS="$(synogroup --get $LEGACY_GROUP | grep '^[0-9]' | sed 's/.*\[\([^]]*\)].*/\1/' | tr '\n' ' ')"
+    if [[ $MEMBERS = *"$LEGACY_USER"* &&  $MEMBERS != *"$NEW_USER"* ]]; then
```

Comment:
Hum. I would take care about such expression which may only run on `bash`... so only on DSM 6. I would check against on DSM 5.2 which runs `busybox` as default shell.

Reply:
So I came to this, because (at least on DSM61) this `if` statement in `postinst` always returns true and the user _always_ gets added even if already present:
```sh
if ! synogroup --get "$GROUP" | grep '^[0-9]:\[${EFF_USER}\]' &> /dev/null; then
```
So something is wrong with this syntax, but I couldn't figure out what. 

I just tested and indeed seems busybox does not support this syntax.. Will come up with a new one!

Reply:
Aah, my bash/shell-noobness that it took me an hour to figure out. It should be of course:
```sh
grep "^[0-9]:\[${EFF_USER}\]"
```
and not
```sh
grep '^[0-9]:\[${EFF_USER}\]'
```
In both commands.